### PR TITLE
worker & atools

### PR DESCRIPTION
**Pull request by gjbex**
* Updated worker quickstart page
* Add FAQ on worker versus atools

### CODE REVIEW

File: source/jobs/worker_framework.rst [hunk 1]
```diff
@@ -126,168 +128,172 @@ It has to contain the names of the variables on the first line, followed
...
-
-   $ qsub -t 1-100 run.pbs
+   $ qsub  -t 1-100  run.pbs
 
 The effect was that rather than 1 job, the user would actually submit
 100 jobs to the queue system (since this puts quite a burden on the
-scheduler, this is precisely why the scheduler doesn't support job
-arrays).
+scheduler, this is precisely why the use of job arrays is somewhat
+discouraged.
```
Comment (smoors) [inline source/jobs/worker_framework.rst]: maybe replace with this (and maybe add a link to the faq 'worker or atools')?
discouraged for very large number of jobs and/or very short jobs.

File: source/jobs/worker_framework.rst [hunk 2]
```diff
@@ -1,76 +1,79 @@
...
+workload.
 
--  many small jobs determined by parameter variations; the scheduler's
-   task is easier when it does not have to deal with too many jobs.
--  job arrays: replace the ``-t`` for array requests; this was an
-   experimental feature provided by the Torque queue system, but it is
-   not supported by Moab, the current scheduler.
+The framework will keep track of the computations that have been
+performed, those that succeeded or failed, and enable you to resume
+computations easily based on that information.  Additionally, It  helps you
```
Comment (smoors) [inline source/jobs/worker_framework.rst]: nit: 2 spaces between `It` and `helps`

File: source/jobs/worker_or_atools.rst [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
++----------------+--------+--------+
+| MPI- based     | no     | yes    |
++----------------+--------+--------+
+
+.. warning::
+
+   Although this might seems to suggest that since atools can deal with all types
+   of work items, it is the best choice, this is definitely not true.
+
+The table makes it clear that MPI application can not be used in work items
```
Comment (smoors) [inline source/jobs/worker_or_atools.rst]: MPI applications

File: source/jobs/worker_or_atools.rst [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+all work items are executed by a single job, so the scheduler is not involved.
+
+On the other side of the spectrum, i.e., work items that take a very long time 
+complete, atools may be the better choice since work items are executed
+independently.  This however depends on the reliability of the infrastructure.
+
+The following table summarizes this.
+
++------------+--------+--------+
+| walltime   | worker | atools |
```
Comment (smoors) [inline source/jobs/worker_or_atools.rst]: maybe change to this:
'walltime of 1 work item'

File: source/jobs/worker_or_atools.rst [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
++============+========+========+
+| < 1 second | \-     | \-\-   |
++------------+--------+--------+
+| < 1 minute | \+     | \-     |
++------------+--------+--------+
+|            | \+\+   | \+\+   |
```
Comment (smoors) [inline source/jobs/worker_or_atools.rst]: 1-24 hours ?

File: source/jobs/worker_or_atools.rst [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+or with many different parameter values.  You can of course manage such
+jobs by hand, or write scripts to do that for you.  However, this is a
+very common scenario, so we developed software to do that for you.
+
+Two general purpose software packages are available:
+
+- the `worker framework <https://github.com/gjbex/worker>`_ (:ref:`worker quickstart <worker framework>` and `worker documentation`_) and
+- `atools <https://github.com/gjbex/atools>`_ (`atools documentation`_).
+
+Both are designed to handle this use case, but with distinct twists.
```
Comment (smoors) [inline source/jobs/worker_or_atools.rst]: maybe replace with this?
Both are designed to handle this use case, but each has its own strengths and weaknesses.

File: source/jobs/worker_or_atools.rst [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+| Leuven   | genius             | gpu       | shared      | yes        |
++----------+--------------------+-----------+-------------+------------+
+| Leuven   | genius             | superdome | shared      | yes        |
++----------+--------------------+-----------+-------------+------------+
+| Leuven   | breniac (Tier-1)   | default   | single job  | yes        |
++----------+--------------------+-----------+-------------+------------+
+
+Clusters with accounting enabled:
+   If you use atools on a cluster where accounting is active, make sure a work
+   item uses all resources of that node.  If multiple work items run on the same
```
Comment (smoors) [inline source/jobs/worker_or_atools.rst]: I guess you mean 'all cores of that node'?

File: source/jobs/worker_or_atools.rst [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+   node concurrently, you will be charged for each work item individually,
+   making that a *very* expensive computation.  In this situation, use worker.
+Clusters with single user policy:
+   Ensure that the load balance is as good as possible.  If a few work items
+   require much more time than others, they may block the nodes they are running
+   on from running other jobs.  This is the case for both atools and worker.
+   However, since worker is an MPI application, it will keep all nodes involved
+   in the job blocked, aggravating the problem.
+Cluster with shared policy
```
Comment (smoors) [inline source/jobs/worker_or_atools.rst]: Clusters with shared policy:

File: source/jobs/worker_or_atools.rst [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+   Here atools allows the scheduler the most flexibility, unless the number of
+   job items is too high.
```
Comment (smoors) [inline source/jobs/worker_or_atools.rst]: job items is too high and/or the jobs are too short.

File: source/jobs/worker_or_atools.rst (hunk 7 above)
Comment (gjbex) [reply source/jobs/worker_or_atools.rst]: No, I don't.  A work item may require more memory than RAM/cores.  Hence all memory would be used, but not all cores.

**Pull request merged**